src/core/symbology/qgssymbol.cpp

```diff
@@ -1560,6 +1560,10 @@ void QgsSymbol::renderFeature( const QgsFeature &feature, QgsRenderContext &cont
       break;
     }
   }
+  if ( clippingEnabled && context.testFlag( Qgis::RenderContextFlag::DisableSymbolClippingToExtent ) )
+  {
+    clippingEnabled = false;
+  }
```

Comment:
```suggestion
  clippingEnabled &= !context.testFlag( Qgis::RenderContextFlag::DisableSymbolClippingToExtent ) 
```
Cosmetic proposal. less code, better readability. We could also group all other conditions related to this boolean, but that's not mandatory.